### PR TITLE
Port User Groups test to AirGun

### PR DESCRIPTION
**Pull request by mirzal**
Port tier2 tests to AirGun.

Requires [AirGun PR #148](https://github.com/SatelliteQE/airgun/pull/148) to be merged.

### CODE REVIEW

**Comment by oshtaier** [comment]
You can create three very simple test cases: `test_positive_create`, `test_positive_update`, `test_positive_delete` just for sanity reasons.
And actually, if you created `test_positive_update` it would catch that typo in airgun

File: tests/foreman/ui_airgun/test_usergroup.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+from robottelo.datafactory import generate_strings_list
+from robottelo.decorators import (
+    parametrize,
+    tier1,
+    tier2,
+    upgrade,
+)
+
+
+@tier1
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_usergroup.py]: remove tier1 decorator as we do not port or introduce tier1 tests. Let's keep all other your changes as it is, just remove decorators

**Comment by oshtaier** [comment]
And do not forget to provide test results console output here. That is mandatory requirements for all our PRs

**Comment by mirzal** [comment]
@oshtaier 

> remove tier1 decorator as we do not port or introduce tier1 tests

But these tests are ported, i.e. they are removed from `tests/foreman/ui/test_usergroup.py`. Should I re-add them to that file, then?

**Comment by oshtaier** [comment]
I just changed my mind. Thanks that you asked that question. UI folder is not used anymore by anyone, but people who port tier1 tests into Minitest suite should see scope of their work. Please revert removed code from UI folder and tier1 decorators from UI_airgun tests, but keep them as it is. Of course, also remove `:id:` information from tier1 tests docstrings

**Comment by mirzal** [comment]
```
$ pytest -v tests/foreman/ui_airgun/test_usergroup.py
============================= test session starts ==============================
platform linux -- Python 3.6.4, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /home/mzalewsk/.local/share/virtualenvs/robottelo-ps2A3-Bl/bin/python3
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/mzalewsk/git/robottelo, inifile:
plugins: wait-for-1.0.9, services-1.2.1, mock-1.6.3
collecting ... collected 22 items
2018-07-11 14:40:03 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui_airgun/test_usergroup.py::test_positive_create_with_name[OrsWhYGLgRvzyjnAZOaxhhRceMjnL] PASSED [  4%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_create_with_name[14550174837808832397809071961] PASSED [  9%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_create_with_name[1nmuoSVD5WAdf5qEnCfaIdSBXDs9F] PASSED [ 13%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_create_with_name[\xcd\xdb\xea\xf1\xd5\xeb\xf0\xfc\xdf\xca\xfe\xf3\xe2\xeb\xfb\xc4\xd3\xf4\xcb\xcd\xd9\xe4\xf8\xed\xeb\xd0\xf0\xca\xde] PASSED [ 18%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_create_with_name[\u37d0\uccf0\uc6e0\U000112c4\u3495\U0001d779\u536f\u62b3\ud5aa\u5cc1\u9a7a\U00018a9d\u31fa\uc22e\ud76d\U000100e3\u857f\ub7c9\u535c\u471c\U00017917\ufb4e\U0001761f\ua1f3\u8cb6\u52e5\ud0f6\uc111\u41b0] FAILED [ 22%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_create_with_name[\u5e18\u9026\u75cf\u80dd\u8c7c\u894e\u4ebb\u7043\u5e70\u98e5\u9515\u9b96\u649c\u8eef\u767c\u6354\u6f14\u5ee6\u9373\u9849\u5246\u57ea\u9b20\u6e15\u6c47\u56b4\u79e4\u9f09\u91cd] PASSED [ 27%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_create_with_name[<button>vgLSllEPmpFMYTLUpyRzRtYzpKzkY</button>] PASSED [ 31%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_update_name[QniSK] PASSED [ 36%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_update_name[76361] PASSED [ 40%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_update_name[qyRbd] PASSED [ 45%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_update_name[\xc5\xe1\xf0\xfa\xf0] PASSED [ 50%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_update_name[\u93f4\ud263\u4e36\u980a\u3485] PASSED [ 54%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_update_name[\u637c\u74c4\u9f9e\u57f8\u6a2d] PASSED [ 59%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_update_name[<div>lEYQa</div>] PASSED [ 63%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_delete_empty[mYLkq] PASSED [ 68%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_delete_empty[43323] PASSED [ 72%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_delete_empty[TTaH9] PASSED [ 77%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_delete_empty[\xe9\xf1\xed\xc4\xee] PASSED [ 81%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_delete_empty[\ud6b1\u0824\uab15\xef\u0538] PASSED [ 86%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_delete_empty[\u6286\u8434\u8f57\u9c2b\u6d69] PASSED [ 90%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_delete_empty[<frame>hBVdM</frame>] FAILED [ 95%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_delete_with_user PASSED [100%]
```

Yes, there are two failures, but they are not caused by airgun, so I am a little hesitant to work around them here.

1. One of `test_positive_create_with_name` failures is caused by ChromeDriver exception "ChromeDriver only supports characters in the BMP". The cause is `generate_strings_list()` function - it invokes fauxfactory `gen_utf8()` that might return valid unicode characters that happens to be outside of plane 0 - and ChromeDriver dislikes them. `gen_utf8` has a flag to limit characters to BMP, but only when invoked directly (i.e. our abstraction layers are not able to set it).
2. One of `test_positive_delete_empty` failures is caused by Satellite ignoring table queries with HTML tags in them - it will just return all records.

Both can go away if I remove parametrization from pytest. But I'm not sure if hiding problems is correct way forward here.

**Comment by oshtaier** [comment]
1) is well known issue. Try `@parametrize('group_name', **valid_data_list('ui'))` instead

**Comment by oshtaier** [comment]
2. Yeah, that is not important, so remove it

**Comment by mirzal** [comment]
After removal of parameters:

```
$ pytest -v tests/foreman/ui_airgun/test_usergroup.py               
===================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.4, pytest-3.4.0, py-1.5.3, pluggy-0.6.0 -- /home/mzalewsk/.local/share/virtualenvs/robottelo-ps2A3-Bl/bin/python3
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/mzalewsk/git/robottelo, inifile:
plugins: wait-for-1.0.9, services-1.2.1, mock-1.6.3
collected 4 items                                                                                                                                                                             
2018-07-11 17:05:35 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui_airgun/test_usergroup.py::test_positive_create_with_name PASSED                                                                                                        [ 25%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_update_name PASSED                                                                                                             [ 50%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_delete_empty PASSED                                                                                                            [ 75%]
tests/foreman/ui_airgun/test_usergroup.py::test_positive_delete_with_user PASSED
```

**Comment by oshtaier** [comment]
of course, remove tier2 from UI folder and do not forget to run `flake8 .` each time you like to submit anything

File: tests/foreman/ui_airgun/test_usergroup.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        organization=[module_org],
+    ).create()
+
+    with session:
+        session.usergroup.create({
+            'usergroup.name': group_name,
+            'usergroup.users': {'assigned': [user_name]},
+        })
+        session.usergroup.delete(group_name)
+        assert session.user.search(user_name) is not None
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_usergroup.py]: also check that usergroup is actually deleted:
`assert not session.usergroup.search(group_name)`

File: tests/foreman/ui_airgun/test_usergroup.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    group_name = gen_string('alpha')
+    new_group_name = gen_utf8(smp=False)
+    with session:
+        session.usergroup.create({
+            'usergroup.name': group_name,
+        })
+        session.usergroup.update(group_name, {
+            'usergroup.name': new_group_name,
+        })
+        assert session.usergroup.search(new_group_name) is not None
```
Comment (ldjebran) [inline tests/foreman/ui_airgun/test_usergroup.py]: This must after create, we have to assert that the new_group_name is  saved

File: tests/foreman/ui_airgun/test_usergroup.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    :CaseImportance: Critical
+    """
+    group_name = gen_utf8(smp=False)
+    with session:
+        session.usergroup.create({
+            'usergroup.name': group_name,
+        })
+        session.usergroup.delete(group_name)
+        assert session.usergroup.search(group_name) == []
```
Comment (ldjebran) [inline tests/foreman/ui_airgun/test_usergroup.py]: we have to assert that group_name is not in session.usergroup.search(group_name)
this way if an other test create a Usergroup, this test will not affected.

File: tests/foreman/ui_airgun/test_usergroup.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+
+@tier2
+@upgrade
+def test_positive_delete_with_user(session, module_org):
+    """Delete an Usergroup that contains a user
```
Comment (ldjebran) [inline tests/foreman/ui_airgun/test_usergroup.py]: Delete a Usergroup that contains a user

**Comment by mirzal** [comment]
@ldjebran re lines 55-58: I'm not sure if I understand your comment correctly. Assertion that name is saved correctly during creation of user group is done in `test_positive_create_with_name`. Do you advise that the same assertion should be repeated in `test_positive_update_name`? I think that check will fail anyway if name is not saved, as update operation works on usergroup selected by name.

**Comment by ldjebran** [comment]
@mirzal , many thanks for your PR, It was OK to search by the new name.

**Pull request merged**